EIPS/eip-5219.md

```diff
@@ -0,0 +1,75 @@
+---
+eip: 5219
+title: Contract REST
+description: Allows the sending of HTTP-like requests to contracts
+author: Pandapip1 (@Pandapip1)
+discussions-to: https://ethereum-magicians.org/t/pr-5219-discussion-contract-rest/9907
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-07-10
+requires: 137, 165, 4804, 4834
+---
+
+## Abstract
+
+Ethereum is the most-established blockchain for building decentralized applications (referred to as `DApp`s). Due to this, the Ethereum DApp ecosystem is very diverse. However, one issue that plagues DApps is the fact that they are not fully decentralized. Specifically, to interface a "decentralized" application, one first needs to access a *centralized* website containing the DApp's front-end code, presenting a few issues (discussed in the Motivation). To solve these issues, this EIP introduces an interface allowing clients to make HTTP-like requests (containing a method, a resource to request, a request body, and headers) to smart contracts, and to receive HTTP-like responses (containing headers and a body).
```

Comment:
This feels like mostly a motivation rather than abstract.  The abstract should describe the specification in a pithy and human readable way, and leave the *why* to the motivation.